app/models/academic.rb

```diff
@@ -25,4 +33,19 @@ def self.search(search)
       order(:name)
     end
   end
+
+  def self.find_for_database_authentication(warden_conditions)
```

Comment:
This method isn't necessary. The authentication_keys: [:ra] do that, it is when you use two fields for authentication, for example login and email.